### PR TITLE
feat: add new component select

### PR DESCRIPTION
**Pull request by nmerget**

### CODE REVIEW

File: packages/components/src/components/select/docs/Angular.md [hunk 1]
```diff
@@ -0,0 +1,24 @@
...
+  ...
+})
+
+```
+
+### Use component
+
+```html app.component.html
+<!-- app.component.html -->
+<db-select>Select</db-select>
```
Comment (mfranzke) [inline packages/components/src/components/select/docs/Angular.md:23]: a string as a children of a `<select>` element wouldn't be the regular use case, but rather `<option>` elements.

File: packages/components/src/components/select/select.scss [hunk 2]
```diff
@@ -0,0 +1,171 @@
...
+	&:has(select:disabled) {
+		// Decided against cursor: not-allowed, compare to e.g. https://phabricator.wikimedia.org/T121960
+		opacity: 0.25;
+		pointer-events: none;
+	}
+
+	label {
+		@extend %db-body-2xs;
+		top: $db-spacing-fixed-2xs;
+		transition: left, top, font-size 0.2s $db-transition-functional-timing;
```
[the remark this answers is not in the document]
Comment (nmerget) [reply packages/components/src/components/select/select.scss:103]: Okay, maybe you can change this for me?
We should use something similar for input and select

File: packages/components/src/components/select/model.ts [hunk 3]
```diff
@@ -0,0 +1,56 @@
...
+	 * If you don't/can't use children/slots you can pass in the options as an array.
+	 */
+	options?: DBSelectOptionType[];
+}
+
+export type DBSelectOptionType = {
+	disabled?: boolean;
+	label?: string;
+	options?: DBSelectOptionType[];
+	value: string;
```
Comment (annsch) [inline packages/components/src/components/select/model.ts:34]: Shouldn't value allow more than just string, e.g. number or object or array?

File: packages/components/src/components/select/model.ts [hunk 4]
```diff
@@ -0,0 +1,56 @@
...
+	 */
+	description?: string;
+
+	/**
+	 * If you don't/can't use children/slots you can pass in the options as an array.
+	 */
+	options?: DBSelectOptionType[];
+}
+
+export type DBSelectOptionType = {
```
Comment (annsch) [inline packages/components/src/components/select/model.ts:30]: These nested Props should also be documented in our props table, should we open an issue for that?

File: packages/components/src/components/select/select.lite.tsx [hunk 5]
```diff
@@ -0,0 +1,181 @@
...
+			<div
+				class={state.getClassNames('db-select', props.className)}
+				data-variant={props.variant}
+				data-icon={props.icon}>
+				<Show when={state.stylePath}>
+					<link rel="stylesheet" href={state.stylePath} />
+				</Show>
+				{/* Required has to be true to use floating label */}
+				<select
+					required
```
Comment (mfranzke) [inline packages/components/src/components/select/select.lite.tsx:104]: `required` isn't necessary for floating labels on selects anymore, compare to https://github.com/db-ui/core/pull/422/commits/23170746cd978d490d5aeeb5a569079da40d9d9e

File: packages/components/src/components/select/docs/Angular.md (hunk 1 above)
Comment (nmerget) [reply packages/components/src/components/select/docs/Angular.md]: fixed

File: packages/components/src/components/select/select.scss (hunk 2 above)
[the remark this answers is not in the document]
Comment (nmerget) [reply packages/components/src/components/select/select.scss:103]: We handle this here: https://github.com/db-ui/mono/issues/908

File: packages/components/src/components/select/model.ts (hunk 4 above)
Comment (nmerget) [reply packages/components/src/components/select/model.ts:30]: fixed

File: packages/components/src/components/select/select.lite.tsx (hunk 5 above)
Comment (nmerget) [reply packages/components/src/components/select/select.lite.tsx]: fixed

File: packages/components/src/components/select/model.ts (hunk 3 above)
Comment (nmerget) [reply packages/components/src/components/select/model.ts]: fixed

**Review by annsch — APPROVED** [review]

**Pull request merged**